lib/utils/report.cjs

```diff
@@ -126,6 +108,42 @@ function report(problem) {
 	result.warn(warningMessage, warningProperties);
 }
 
+/**
+ * @param {import('stylelint').Problem} problem
+ * @return {problem is import('stylelint').ProblemWithNormalizedPositions}
+ */
+function normalizeReportSourceLocation(problem) {
+	// Use the given node to calculate the start and end positions.
+	if (problem.node && (!problem.start || !problem.end)) {
+		const range = nodeRangeBy(problem.node, { index: problem.index, endIndex: problem.endIndex });
+
+		if (range) {
+			problem.start ??= range.start;
+			problem.end ??= range.end;
+		}
+	}
+
+	if (!validateTypes.isNumber(problem.line) && validateTypes.isNumber(problem.start?.line)) {
```

Comment:
nit
question

No one will pass it, since line always starts at `1`, but this allows `0`.
What about just `(!problem.line && problem.start?.line)`?

Reply:
See: https://github.com/stylelint/stylelint/pull/8098#discussion_r1824063826

Reply:
If passing `line === 0` is a hard error then maybe we should throw an actual error?
That would be a separate PR, but it seems reasonable to me to do such validation.

Reply:
```suggestion
	if (!problem.line && problem.start?.line) {
```